src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/ClassMemberImpliedModifierCheckTest.java

```diff
@@ -0,0 +1,134 @@
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+public class ClassMemberImpliedModifierCheckTest
+    extends AbstractModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/checks/modifier/classmemberimpliedmodifier";
+    }
+
+    @Test
+    public void testMethodsOnClass() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(ClassMemberImpliedModifierCheck.class);
+        final String[] expected = {
+            "42:9: " + getCheckMessage(MSG_KEY, "static"),
+            "49:9: " + getCheckMessage(MSG_KEY, "static"),
+            "53:5: " + getCheckMessage(MSG_KEY, "static"),
+            "60:9: " + getCheckMessage(MSG_KEY, "static"),
+            "67:9: " + getCheckMessage(MSG_KEY, "static"),
+            "74:5: " + getCheckMessage(MSG_KEY, "static"),
+        };
+        verify(checkConfig, getPath("InputClassMemberImpliedModifierOnClass.java"),
+            expected);
+    }
+
+    @Test
+    public void testMethodsOnClassNoImpliedStaticEnum() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(ClassMemberImpliedModifierCheck.class);
+        checkConfig.addAttribute("violateImpliedStaticOnNestedEnum", "false");
+        final String[] expected = {
+            "50:9: " + getCheckMessage(MSG_KEY, "static"),
+            "68:9: " + getCheckMessage(MSG_KEY, "static"),
+            "75:5: " + getCheckMessage(MSG_KEY, "static"),
+        };
+        verify(checkConfig, getPath("InputClassMemberImpliedModifierOnClassSetEnumFalse.java"),
+            expected);
+    }
+
+    @Test
+    public void testMethodsOnClassNoImpliedStaticInterface() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(ClassMemberImpliedModifierCheck.class);
+        checkConfig.addAttribute("violateImpliedStaticOnNestedInterface", "false");
+        final String[] expected = {
+            "43:9: " + getCheckMessage(MSG_KEY, "static"),
+            "54:5: " + getCheckMessage(MSG_KEY, "static"),
+            "61:9: " + getCheckMessage(MSG_KEY, "static"),
+        };
+        verify(checkConfig, getPath("InputClassMemberImpliedModifierOnClassSetInterfaceFalse.java"),
+            expected);
+    }
+
+    @Test
+    public void testMethodsOnClassNoViolationsChecked() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(ClassMemberImpliedModifierCheck.class);
+        checkConfig.addAttribute("violateImpliedStaticOnNestedEnum", "false");
+        checkConfig.addAttribute("violateImpliedStaticOnNestedInterface", "false");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputClassMemberImpliedModifierOnClass.java"),
```

Comment:
Please do not reused the same input file  in different  test methods.
Sorry for being too demanding.

Reply:
Done